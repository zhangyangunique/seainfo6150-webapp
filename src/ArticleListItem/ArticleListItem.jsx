import React from "react";
import PropTypes from "prop-types";

const ArticleListItem = props => {
	return (
		<article>
			<h2>{props.article.title}</h2>
			<time dateTime={props.article.timeStamp}>{props.article.displayDate}</time>
			<p>{props.article.shortText}</p>
		</article>
	);
};

ArticleListItem.propTypes = {
	article: PropTypes.object.isRequired
};

export default ArticleListItem;