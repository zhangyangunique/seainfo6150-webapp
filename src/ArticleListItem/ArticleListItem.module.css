.container {
  background-image: linear-gradient(to right, rgba(204, 0, 0, 0.5), #eee 20%);
  border: 1px solid #c00;
  border-radius: 10px;
  box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.3);
  font-family: Arial, Helvetica, sans-serif;
  padding: 30px;
}

.article {
  display: flex;
  flex-flow: column wrap;
  justify-content: space-between;
  height: 100%;
}

.wrapper {
  flex: 1;
}

.title {
  color: #c00;
  font-size: 2em;
  margin: 0 0 20px;
  text-decoration: underline;
}

.text {
  font-size: 12px;
  margin: 0 0 20px;
}

.time {
  font-weight: bold;
}

@media screen and (min-width: 525px) {
  .container {
    padding: 20px;
  }
  .title {
    font-size: 1.5em;
  }
}

@media screen and (min-width: 800px) {
  .container {
    padding: 15px;
  }
  .title {
    font-size: 1em;
  }
}

@media screen and (min-width: 1150px) {
  .container {
    padding: 10px;
  }
  .title {
    font-size: 0.8em;
  }
}
