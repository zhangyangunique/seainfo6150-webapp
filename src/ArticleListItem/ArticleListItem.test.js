import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";

const article = {
	"slug": "181122102925-statue-of-liberty-crown",
	"title": "The Statue of Liberty's torch heads to new museum",
	"shortText": "The Statue of Liberty's original torch is getting a new home",
	"text": "<p>The Statue of Liberty's original torch is getting a new home.</p><p>Made of copper and gold so that it would shine bright, the first torch took a beating from the weather and was replaced by a replica in 1984. It's been stored in the statue's pedestal ever since.</p><p>But visitors will be able to see the original 3,600-pound torch in all its glory at a new Statue of Liberty Museum, which will open on Liberty Island in May 2019. The torch was transported there by truck last week, along with a replica model of Lady Liberty's face.</p><p><q>Although it is not one of the most difficult things we have ever moved,</q> says Douglas Phelps, who oversaw the relocation of the torch, <q>it is certainly the most important.</q><p>Not only is the Statue of Liberty the symbol of New York City, she is also a UNESCO World Heritage site under the designation of &ldquo;Outstanding Universal Value.&rdquo;</p><p>Her full name is Liberty Enlightening the World, and she was designed by Frédéric Auguste Bartholdi and built by Gustave Eiffel -- yes, the same guy who built Paris's Eiffel Tower.</p><p>Lady Liberty was a gift from France to the United States to celebrate the centennial of their nation's independence -- although, somewhat awkwardly, she arrived a decade too late. Nevertheless, President Grover Cleveland formally unveiled her in 1886.</p><p>Since then, the statue has served as a symbol of hope and possibility for many people who entered the United States as immigrants via neighboring Ellis Island.</p><p><q>The original torch is a beautiful object of art and exemplifies the American ideals of freedom and democracy,</q> says John Piltzecker, superintendent of the Statue of Liberty National Monument. That symbolic spirit was bolstered by the decision to inscribe Emma Lazarus' poem &ldquo;The New Colossus&rdquo; on the statue's pedestal, which gave Lady Liberty one of her nicknames, &ldquo;Mother of Exiles.&rdquo;</p><p>The most famous lines in the poem are: &ldquo;Give me your tired, your poor / Your huddled masses yearning to breathe free.&rdquo;</p><p>In 2017, nearly 4.5 million people visited Liberty Island.</p><p>When completed, the Statue of Liberty Museum will be 26,000 square feet of history, inspiration and deeper analysis of the statue's symbolism both in the US and around the world.</p><p>The &ldquo;inspiration gallery,&rdquo; which is where the torch will live, will have floor-to-ceiling glass windows looking out at Lower Manhattan, including the World Trade Center.</p>",
	"pubDate": "Thu, 22 Nov 2018 15:12:24 GMT",
	"pubYear": "2018",
	"author": "Kate Farley",
	"authorEmail": "kate.farley@nytimes.com",
	"image": {
		"_url": "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg",
		"_height": "619",
		"_width": "1100"
	},
	"timeStamp": "2018-11-22T15:12:24.000Z",
	"displayDate": "November 22nd 2018, 7:12 am"
};

describe("ArticleListItem tests", () => {
	it("renders correctly", () => {
		const { container } = render(<ArticleListItem article={article} />);
		expect(container).toMatchSnapshot();
	});

	it("renders correctly after button is clicked", () => {
		const { container, getByText } = render(<ArticleListItem article={article} />);

		const showMoreButton = getByText("Show more");
		fireEvent.click(showMoreButton);
		expect(container).toMatchSnapshot();

		const showLessButton = getByText("Show less");
		fireEvent.click(showLessButton);
		expect(container).toMatchSnapshot();
	});
}); 