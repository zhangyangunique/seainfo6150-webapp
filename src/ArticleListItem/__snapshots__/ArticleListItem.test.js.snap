// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <li
    class="container"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <img
          alt="The Statue of Liberty's torch heads to new museum"
          class="image"
          src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
        />
        <h2
          class="title"
        >
          The Statue of Liberty's torch heads to new museum
        </h2>
      </div>
      <button
        class="button"
      >
        Show more
      </button>
    </article>
  </li>
</div>
`;
